subsys/net/ip/ipv6_nbr.c

```diff
@@ -2393,6 +2393,10 @@ static enum net_verdict handle_ra_input(struct net_pkt *pkt,
 
 	nd_opt_hdr = (struct net_icmpv6_nd_opt_hdr *)
 				net_pkt_get_data(pkt, &nd_access);
+
+	/* Add neighbor based on link local address received in Router Advertisement */
+	nbr = add_nbr(net_pkt_iface(pkt), (struct in6_addr *)NET_IPV6_HDR(pkt)->src, true, NET_IPV6_NBR_STATE_INCOMPLETE);
```

Comment:
`add_nbr` will not trigger the `NET_EVENT_IPV6_NBR_ADD` event, it seems inconsistent with the "regular" neighbor addition. Perhaps `net_ipv6_nbr_add()` should be updated to allow NULL LL address and used instead.

I'd also move the neighbour addition **after** options have been processed, and only add an entry if the `SLLAO` option was not present (`nbr` pointer NULL).

Reply:
> add_nbr will not trigger the NET_EVENT_IPV6_NBR_ADD event, it seems inconsistent with the "regular" neighbor addition. Perhaps net_ipv6_nbr_add() should be updated to allow NULL LL address and used instead.

Ok, I replaced `add_nbr()` with `net_ipv6_nbr_add()` and it worked fine for NULL, so I think it even doesn't need to be updated.

> I'd also move the neighbour addition after options have been processed, and only add an entry if the SLLAO option was not present (nbr pointer NULL).

Hmm I think it will not solve the problem raised in this PR, as handling `NET_ICMPV6_ND_OPT_ROUTE` option results in trying to add new route in `net_route_add()` method. If we will not create NBR first `net_route_add()` will fail, because `net_ipv6_nbr_lookup()` will return false.

Reply:
> Ok, I replaced add_nbr() with net_ipv6_nbr_add() and it worked fine for NULL, so I think it even doesn't need to be updated.

There are places where `lladdr` is dereferenced (https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/ipv6_nbr.c#L595 for example) so a few additional checks are needed in there.

> Hmm I think it will not solve the problem raised in this PR, as handling NET_ICMPV6_ND_OPT_ROUTE option results in trying to add new route in net_route_add() method. If we will not create NBR first net_route_add() will fail, because net_ipv6_nbr_lookup() will return false.

Ok, thanks for clarifying. Still not a fan of calling `nbr_add()` unconditionally, but I see no better option given this argument...

Reply:
> There are places where lladdr is dereferenced (https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/ipv6_nbr.c#L595 for example) so a few additional checks are needed in there.

Right, I hope it should be addressed now.